src/controllers/user.controller.ts

```diff
@@ -0,0 +1,57 @@
+import { randomUUID } from 'crypto';
+
+import { BaseUser, User } from '../models/user';
+import { users } from '../data/data';
+
+export const getUserById = (id: string) => users.find((item) => item.id === id);
+
+export const saveUser = (user: User) => {
```

Comment:
why don't you save inside `create` function? inside `update` you mutate `users` list so why couldn't do the same here?